OemPkg/OemConfigPolicyCreatorPei/OemConfigPolicyCreatorPei.c

```diff
@@ -0,0 +1,248 @@
+/** @file
+  This module receives the static platform data for a platform and creates a config policy. It is intended to run
+  before a platform level Silicon Policy Creator that maps this config policy to silicon policy.
+
+  Copyright (c) Microsoft Corporation.
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+**/
+
+#include <Uefi.h>
+
+#include <Ppi/Policy.h>
+#include <Library/DebugLib.h>
+#include <Library/BaseMemoryLib.h>
+#include <Library/PeiServicesLib.h>
+#include <Library/DebugLib.h>
+#include <Library/MemoryAllocationLib.h>
+#include <Guid/VariableFormat.h>
+#include <Library/ConfigVariableListLib.h>
+#include <Library/ConfigKnobShimLib.h>
+#include <Library/SafeIntLib.h>
+#include <ConfigStdStructDefs.h>
+
+extern KNOB_DATA  gKnobData[];
+extern UINTN      gNumKnobs;
+
+/**
+  Helper function to create config policy.
+
+  @param[out]      ConfPolicy     Pointer to uninitialized config policy
+  @param[out]      ConfPolicySize Pointer to size of created ConfPolicy
+
+  @retval EFI_SUCCESS           The configuration is translated to policy successfully.
+  @retval EFI_OUT_OF_RESOURCES  Memory allocation failed.
+  @retval EFI_NOT_READY         Variable Services were not found.
+  @retval EFI_ABORTED           Creating variable list failed.
+  @retval Others                Other errors occurred when getting GFX policy.
+**/
+STATIC
+EFI_STATUS
+CreateConfPolicy (
+  OUT  VOID     **ConfPolicy,
+  OUT   UINT16  *ConfPolicySize
+  )
+{
+  EFI_STATUS             Status;
+  UINT32                 i;
+  UINT32                 NeededSize = 0;
+  UINT32                 Offset     = 0;
+  CHAR16                 UnicodeName[CONF_VAR_NAME_LEN];           // get a buffer of the max name size
+  CONFIG_VAR_LIST_ENTRY  VarListEntry;
+  UINTN                  VarListSize;
+  VOID                   *ConfListPtr;
+  UINT32                 UnicodeNameSize;
+  UINT32                 TmpNeededSize;
+
+  // first figure out how much space we need to allocate for the ConfPolicy
+  for (i = 0; i < gNumKnobs; i++) {
+    Status = (EFI_STATUS)SafeUint32Mult (gKnobData[i].NameSize, 2, &UnicodeNameSize);
+    if (EFI_ERROR (Status)) {
+      // we overflowed
+      DEBUG ((DEBUG_ERROR, "%a knob %s has too long a name!\n", __FUNCTION__, gKnobData[i].Name));

```

Comment:
Do you just want to print the name size instead? The name content might be bogus if the size is overflowing.